ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ExpiryDateValidationResult.kt

```diff
@@ -11,9 +11,9 @@ package com.adyen.checkout.ui.core.internal.util
 import androidx.annotation.RestrictTo
 
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-enum class ExpiryDateValidationResult {
-    VALID,
-    INVALID_TOO_FAR_IN_THE_FUTURE,
-    INVALID_TOO_OLD,
-    INVALID_EXPIRY_DATE
+sealed class ExpiryDateValidationResult {
+    class Valid : ExpiryDateValidationResult()
+    class InvalidTooFarInTheFuture : ExpiryDateValidationResult()
+    class InvalidTooOld : ExpiryDateValidationResult()
+    data class InvalidExpiryDate(val isDateFormatInvalid: Boolean) : ExpiryDateValidationResult()
```

Comment:
What is `isDateFormatInvalid` used for?

Reply:
Expiry date can be invalid due to it being empty, misformatted or partially formatted. We need to check this in `CardValidationUtils:105`.

Reply:
What I could do is create a different enum value for the specific case when isDateFormatInvalid is true to keep it more consistent and not pass any arguments.

Reply:
I think that would be more clear